### PR TITLE
FEATURE: BB-1712 more friendly MVF

### PR DESCRIPTION
**Pull request by GALLLASMILAN**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by GALLLASMILAN** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1298.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5473/ : SUCCESS in 9m 58s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/971/ : SUCCESS in 10s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/972/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2254/ : SUCCESS in 20m 59s

File: src/components/filters/MeasureValueFilter/OperatorDropdownItem.tsx [hunk 1]
```diff
@@ -48,6 +55,23 @@ export class OperatorDropdownItem extends React.PureComponent<IOperatorDropdownI
         onClick(operator);
         e.preventDefault();
     };
+
+    private renderBubble(message: string) {
+        return (
+            <div className="tooltip-bubble">
+                <BubbleHoverTrigger tagName={"div"} showDelay={400} hideDelay={200}>
+                    {this.renderTooltip()}
```
Comment (strouhaljan) [inline src/components/filters/MeasureValueFilter/OperatorDropdownItem.tsx]: There is no need to put it to separate function in my opinion.

File: src/components/filters/MeasureValueFilter/OperatorDropdownItem.tsx [hunk 2]
```diff
@@ -5,12 +5,16 @@ import classNames from "classnames";
...
 import { string as stringUtils } from "@gooddata/js-utils";
+import BubbleHoverTrigger from "@gooddata/goodstrap/lib/Bubble/BubbleHoverTrigger";
+import Bubble from "@gooddata/goodstrap/lib/Bubble/Bubble";
 
 import { getOperatorTranslationKey, getOperatorIcon } from "../../../helpers/measureValueFilterOperator";
 
 export interface IOperatorDropdownItemOwnProps {
     selectedOperator: string;
     operator: string;
+    showBuble?: boolean;
```
Comment (strouhaljan) [inline src/components/filters/MeasureValueFilter/OperatorDropdownItem.tsx]: Maybe we could skip the `showBubble` property and just turn it on/off by setting `bubbleText` property?

File: src/components/filters/MeasureValueFilter/DropdownBody.tsx [hunk 3]
```diff
@@ -54,7 +54,10 @@ class DropdownBodyWrapped extends React.PureComponent<IDropdownBodyProps, IDropd
                     <div className="gd-mvf-dropdown-section">
                         <OperatorDropdown onSelect={this.handleOperatorSelection} operator={operator} />
                     </div>
-                    <div className="gd-mvf-dropdown-section">{this.renderInputSection()}</div>
+
+                    {operator !== "ALL" && (
```
Comment (strouhaljan) [inline src/components/filters/MeasureValueFilter/DropdownBody.tsx]: There is a constant for `ALL` operator.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: src/components/filters/MeasureValueFilter/DropdownBody.tsx (hunk 3 above)
Comment (GALLLASMILAN) [reply src/components/filters/MeasureValueFilter/DropdownBody.tsx]: ok

File: src/components/filters/MeasureValueFilter/OperatorDropdownItem.tsx (hunk 2 above)
Comment (GALLLASMILAN) [reply src/components/filters/MeasureValueFilter/OperatorDropdownItem.tsx]: ok, the only bubbleText property is used now

File: src/components/filters/MeasureValueFilter/OperatorDropdownItem.tsx (hunk 1 above)
Comment (GALLLASMILAN) [reply src/components/filters/MeasureValueFilter/OperatorDropdownItem.tsx]: ok

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5481/ : SUCCESS in 9m 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/977/ : SUCCESS in 10s

File: src/translations/en-US.json [hunk 4]
```diff
@@ -834,5 +834,15 @@
         "value": "not between",
         "comment": "",
         "limit": 0
+    },
+    "mvf.operator.between.tooltip.buble": {
```
Comment (strouhaljan) [inline src/translations/en-US.json]: Typo `bubble`.

File: src/components/filters/MeasureValueFilter/tests/MeasureValueFilter.spec.tsx [hunk 5]
```diff
@@ -69,6 +69,33 @@ describe("Measure value filter", () => {
         ).toEqual(true);
     });
 
+    describe("tooltip", () => {
```
Comment (strouhaljan) [inline src/components/filters/MeasureValueFilter/tests/MeasureValueFilter.spec.tsx]: You can use a parametrized test here.

File: src/components/filters/MeasureValueFilter/OperatorDropdownItem.tsx [hunk 6]
```diff
@@ -48,6 +53,19 @@ export class OperatorDropdownItem extends React.PureComponent<IOperatorDropdownI
         onClick(operator);
         e.preventDefault();
     };
+
+    private renderBubble(message: string) {
```
Comment (strouhaljan) [inline src/components/filters/MeasureValueFilter/OperatorDropdownItem.tsx]: You can pick `bubbleText` property from `props` here, not needed to pass it from `render` method.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: src/components/filters/MeasureValueFilter/tests/MeasureValueFilter.spec.tsx (hunk 5 above)
Comment (GALLLASMILAN) [reply src/components/filters/MeasureValueFilter/tests/MeasureValueFilter.spec.tsx]: ok

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2835/ : SUCCESS in 15m 32s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5488/ : SUCCESS in 7m 58s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1337/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1831/ : SUCCESS in 19m 46s

**Pull request merged**